### PR TITLE
Merge dev into main

### PR DESCRIPTION
**Pull request by nightjar0**
## screen helpers done

### CODE REVIEW

File: commonutils/src/main/java/com/asgatech/commonutils/ImageLoaderHelper.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Image Loading Helpers
+ */
+class ImageLoaderHelper {
+    /**
+     * Load Image From Url
+     */
+    fun loadImage(activity: Context?, imageView: ImageView?, url: String?) {
+        if (url == null || url.isEmpty() || imageView == null) return
```
Comment (IbrahimAli2017) [inline commonutils/src/main/java/com/asgatech/commonutils/ImageLoaderHelper.kt]: replace (url==null || url.isEmpty ) with url.isNullOrEmpty

File: commonutils/src/main/java/com/asgatech/commonutils/ImageLoaderHelper.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.widget.ImageView
+import com.bumptech.glide.Glide
+
+/**
+ * Image Loading Helpers
+ */
+class ImageLoaderHelper {
```
Comment (IbrahimAli2017) [inline commonutils/src/main/java/com/asgatech/commonutils/ImageLoaderHelper.kt:13]: you should add method that take place Holder

File: commonutils/src/main/java/com/asgatech/commonutils/ScreenHelpers.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        @Suppress("DEPRECATION")
+        activity.window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+        activity.window.statusBarColor = Color.TRANSPARENT
+    }
+
+    /**
+     * Set the full screen mode for devices that have R version or later
+     */
+    @RequiresApi(Build.VERSION_CODES.R)
+    private fun setFullScreenForRAndLater(activity: Activity) {
```
Comment (IbrahimAli2017) [inline commonutils/src/main/java/com/asgatech/commonutils/ScreenHelpers.kt]: remove R from the name of function

File: commonutils/src/main/java/com/asgatech/commonutils/WebViewHelper.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ * WebView Helpers to handle showing of content inside WebView in different languages
+ */
+class WebViewHelper {
+    private val SCROLL_DELAY = 50L  // the delay time before scrolling the webView
+
+    /**
+     * Build the WebView with specifying if its rtl or ltr language
+     */
+    fun setWebViewHtml(webView: WebView, htmlString: String?, isRtlLang: Boolean) {
+        if (htmlString == null) return
```
Comment (IbrahimAli2017) [inline commonutils/src/main/java/com/asgatech/commonutils/WebViewHelper.kt]: nullOrEmpty

File: commonutils/src/main/java/com/asgatech/commonutils/permissions/PermissionHandler.kt [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        val selfPermission = ContextCompat.checkSelfPermission(context, permission)
+        return selfPermission == PackageManager.PERMISSION_GRANTED
+    }
+
+    /**
+     * Check if certain permissions are granted or not.
+     */
+    fun isPermissionGranted(context: Context, permissions: Array<String>): Boolean {
+        val grantedOnes: MutableList<String> = ArrayList()
+        for (permission in permissions) {
```
Comment (IbrahimAli2017) [inline commonutils/src/main/java/com/asgatech/commonutils/permissions/PermissionHandler.kt]: you can optimize the performance by check if isPErmissionGranted == false then return false

and replace grantedOnes.size == permissions.size with true

File: commonutils/src/main/java/com/asgatech/commonutils/permissions/PermissionHandler.kt [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    /**
+     * Ask user to grant certain permission
+     */
+    fun askForPermission(
+        context: AppCompatActivity,
+        permission: String,
+        permissionCallback: PermissionsCallback
+    ) {
+        this.permissionCallback = permissionCallback
+        ActivityCompat.requestPermissions(
```
Comment (IbrahimAli2017) [inline commonutils/src/main/java/com/asgatech/commonutils/permissions/PermissionHandler.kt]: call askForPermissions function

**Comment by nightjar0** [comment]
Comments done

**Pull request merged**